apm-agent-plugins/apm-quartz-job-plugin/src/main/java/co/elastic/apm/agent/quartz/job/JobTransactionNameAdvice.java

```diff
@@ -0,0 +1,82 @@
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.quartz.job;
+
+import java.lang.reflect.InvocationTargetException;
+
+import javax.annotation.Nullable;
+
+import org.quartz.JobExecutionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.bci.bytebuddy.SimpleMethodSignatureOffsetMappingFactory.SimpleMethodSignature;
+import co.elastic.apm.agent.impl.transaction.TraceContext;
+import co.elastic.apm.agent.impl.transaction.TraceContextHolder;
+import co.elastic.apm.agent.impl.transaction.Transaction;
+import net.bytebuddy.asm.Advice;
+
+public class JobTransactionNameAdvice {
+    @VisibleForAdvice
+    public static final Logger logger = LoggerFactory.getLogger(JobTransactionNameInstrumentation.class);
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    private static void setTransactionName(@Advice.Argument(readOnly = true, value = 0) JobExecutionContext context,
+    		@SimpleMethodSignature String signature, @Advice.Origin Class<?> clazz, @Advice.Local("transaction") Transaction transaction) 
+    		throws InvocationTargetException, NoSuchMethodException {
+        if (ElasticApmInstrumentation.tracer != null) {
+            TraceContextHolder<?> active = ElasticApmInstrumentation.tracer.getActive();
+            if(context == null) {
+            	logger.warn("Cannot correctly name transaction for method {} because JobExecutionContext is null", signature);
+            	transaction = ElasticApmInstrumentation.tracer.startTransaction(TraceContext.asRoot(), null, clazz.getClassLoader())
+	                    .withName(signature)
+	                    .withType(JobTransactionNameInstrumentation.TRANSACTION_TYPE)
+	                    .activate();
+            }else if (active == null) {
+				transaction = ElasticApmInstrumentation.tracer.startTransaction(TraceContext.asRoot(), null, clazz.getClassLoader())
+	                    .withName(context.getJobDetail().getKey().toString())
+	                    .withType(JobTransactionNameInstrumentation.TRANSACTION_TYPE)
+	                    .activate();
+				transaction.addCustomContext("trigger", context.getTrigger().getKey().toString());
+				transaction.addCustomContext("scheduledTime", context.getScheduledFireTime().toString());
```

Comment:
Quartz separates the logic of when a piec of code is being executed(trigger) from the code itself(job) so you may have multiple trigger executing the same job. In large applications it can be difficult to check why a job was wrongly executed, therefore I want to add the trigger name which triggered the execution of the job. Here, searching for the field might make sense.
As the quartz framework works with a thread pool it can occur that the the pool is empty as all threads are used so a job execution is delayed. In this case the timestamp of the transaction in APM would be the date and time when the job was executed and the field `scheduledTime` would indicate the date and time the job **should** have been executed.